src/hydro_system.hpp

```diff
@@ -826,8 +858,10 @@ void HydroSystem<problem_t>::ComputeFluxes(amrex::MultiFab &x1Flux_mf, amrex::Mu
 				// (pressure_index is actually eint)
 				const double eint_L = x1LeftState(i, j, k, pressure_index);
 				const double eint_R = x1RightState(i, j, k, pressure_index);
-				P_L = rho_L * eint_L * (gamma_ - 1.0);
-				P_R = rho_R * eint_R * (gamma_ - 1.0);
+				amrex::GpuArray<Real, nmscalars_> massScalars = RadSystem<problem_t>::ComputeMassScalars(x1LeftState, i, j, k);
+				P_L = quokka::EOS<problem_t>::ComputePressure(rho_L, eint_L * rho_L, massScalars);
+				massScalars = RadSystem<problem_t>::ComputeMassScalars(x1RightState, i, j, k);
+				P_R = quokka::EOS<problem_t>::ComputePressure(rho_R, eint_R * rho_R, massScalars);
```

Comment:
@BenWibking the HydroContact test is failing because of this modification. The pressure returned by microphysics is initially exactly unity, but then as time goes on it starts slowly deviating away from being exactly 1. by the end of the simulation, it is quite far from 1 (e.g., pressures at the end step are 0.99000 or 1.002, etc.). This test has a zero tolerance because the solution should be exact. 

I wonder if it changing these lines but not updating the Riemann solver makes the input rho and eint deviate from their exact values, which causes microphysics to return a non-exact pressure value? This test passes if I keep all other modifications but revert back to the original code for here.

Reply:
No, it should give exactly the same result here. This might be due to a subtle roundoff error.

Reply:
Can you run the first timestep through the debugger and output `P_R - P_L`? It should be *exactly* zero. If it's some tiny (but non-zero) number, it's a numerical roundoff issue that should be fixed.

Reply:
Yes, `P_R - P_L` is exactly 0 at the first timestep, but slowly starts deviating from 0 as time progresses.

Reply:
Okay, I have narrowed down the problem to Microphysics. If I use the _exact_ same equation in Microphysics as the one in Quokka, I get the exact solution right and the test passes.

To calculate pressure, Microphysics first finds the gas temperature (from eint), and uses the temperature to find pressure. There are a few cancellations that occur in this indirect way of calculating pressure from eint. My guess is that these cancellations are introducing a rounding off error. But, this error is not there in the beginning (at the first step), and starts to emerge over the course of the simulation.

Reply:
FOUND IT! You will not believe what it was.